In [1]:
import pandas as pd

In [2]:
df = pd.read_excel('Data/Unlinked Metadata 0625.xlsx')
df.head()

Subject ID Patient Consented Sample Type_#   CGA  DOL   Current Weight  \
0    NB00012                 N   NB00012_M_1  28.4    27            1430   
1    NB00003                 Y   NB00003_M_1  29.0    22             995   
2    NB00003                 Y   NB00003_M_2  29.0    22            1005   
3    NB00003                 Y   NB00003_M_3  29.3    25            1065   
4    NB00354                 N   NB00354_M_1  29.4    12            1005   

   Current Height  Current HC Scavenged/Fresh?   MBM/DMB? HMF Y/N? TPN Y/N?  \
0            32.0        26.0       Scavenged   MBM + DBM        Y      NaN   
1            35.7        23.7       Scavenged         DBM        Y        N   
2            36.0        24.4       Scavenged         DBM        Y        N   
3            36.0        24.3       Scavenged         DBM        Y        N   
4            34.0        34.5       Scavenged         MBM        Y        Y   

  Iron Y/N?  Linked?  # Aliquots                      Additional Comments  
0        NaN        N    5 (~0.8)  Residual from Milk Prep Room & B2 Full  
1          Y        N           2                  Scavenged Feeding Tube  
2          Y        N           1                  Scavenged Feeding Tube  
3          Y        N           1                  Scavenged Feeding Tube  
4        NaN        N           7            Residual from Milk Prep Room

In [3]:
AC = pd.read_excel('Data/NeoBank Unlinked AC.xlsx')
AC.head()

sample ID  inj vol      2'FL     DFLAC      3'SL      6'SL       LNT  \
0  NB00003_M_27       15       NaN       NaN       NaN       NaN       NaN   
1  NB00003_M_26       15  0.149068  0.025539  0.036240  0.048252  0.032744   
2   NB00010_M_4       15  3.202351  0.211063  0.202066  0.793208  1.288569   
3   NB00004_M_4       15  4.533413  0.015644  0.109186  0.516448  0.978176   
4   NB00014_M_1       15  6.616323  0.997031  0.286334  0.743593  1.071591   

       LNnT     LNFPI    LNFPII   LNFPIII      LSTc     DFLNT     DSLNT  \
0       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
1  0.015477  0.037367  0.029173  0.014387  0.029336  0.024442  0.024050   
2  0.099609  0.896822  0.838056  0.026299  0.198385  0.816358  0.293375   
3  0.235237  2.135244  0.025100  0.028135  0.273544  0.066877  0.159376   
4  0.337355  2.218069  0.367415  0.051010  0.659550  0.795395  0.261531   

      DFLNH    FDSLNH     DSLNH  
0       NaN       NaN       NaN  
1  0.014162  0.024403  0.014275  
2  0.064694  0.116699  0.158850  
3  0.057803  0.033015  0.069969  
4  0.038317  0.048243  0.148319

In [4]:
df = df.rename(columns={"Sample Type_#": "sample_unique_id"})
AC = AC.rename(columns={"sample ID": "sample_unique_id"})


In [5]:
df['sample_unique_id'] = df['sample_unique_id'].str.strip()
AC['sample_unique_id'] = AC['sample_unique_id'].str.strip()

In [6]:
AC

sample_unique_id  inj vol      2'FL     DFLAC      3'SL      6'SL  \
0       NB00003_M_27       15       NaN       NaN       NaN       NaN   
1       NB00003_M_26       15  0.149068  0.025539  0.036240  0.048252   
2        NB00010_M_4       15  3.202351  0.211063  0.202066  0.793208   
3        NB00004_M_4       15  4.533413  0.015644  0.109186  0.516448   
4        NB00014_M_1       15  6.616323  0.997031  0.286334  0.743593   
..               ...      ...       ...       ...       ...       ...   
130      NB00380_M_5       15  0.304270  0.011897  0.263096  0.820189   
131      NB00438_M_3       15  4.197056  0.353557  0.210112  0.510462   
132      NB00457_M_1       15  7.213692  0.555859  0.342633  1.165302   
133      NB00468_M_1       15  3.254378  0.394317  0.246220  0.331024   
134        NB00457_2       15  4.254712  0.361192  0.182488  1.674372   

          LNT      LNnT     LNFPI    LNFPII   LNFPIII      LSTc     DFLNT  \
0         NaN       NaN       NaN       NaN       NaN       NaN       NaN   
1    0.032744  0.015477  0.037367  0.029173  0.014387  0.029336  0.024442   
2    1.288569  0.099609  0.896822  0.838056  0.026299  0.198385  0.816358   
3    0.978176  0.235237  2.135244  0.025100  0.028135  0.273544  0.066877   
4    1.071591  0.337355  2.218069  0.367415  0.051010  0.659550  0.795395   
..        ...       ...       ...       ...       ...       ...       ...   
130  1.305836  0.335492  0.132126  1.380412  0.017233  0.378374  0.178397   
131  0.837567  0.176720  1.504402  0.391415  0.019311  0.300946  0.587013   
132  2.305625  0.246715  3.471317  0.730289  0.109193  0.914475  1.595734   
133  0.403736  0.098364  0.497896  0.573353  0.019541  0.130040  0.511980   
134  1.452224  0.177745  2.317093  0.572663  0.062065  0.629622  1.104219   

        DSLNT     DFLNH    FDSLNH     DSLNH  
0         NaN       NaN       NaN       NaN  
1    0.024050  0.014162  0.024403  0.014275  
2    0.293375  0.064694  0.116699  0.158850  
3    0.159376  0.057803  0.033015  0.069969  
4    0.261531  0.038317  0.048243  0.148319  
..        ...       ...       ...       ...  
130  0.149128  0.099998  0.246744  0.249289  
131  0.103855  0.043828  0.038037  0.116563  
132  0.311655  0.048459  0.029148  0.162687  
133  0.100318  0.066628  0.068355  0.069158  
134  0.206848  0.042000  0.040911  0.146954  

[135 rows x 17 columns]

In [7]:
merged = pd.merge(df, AC, on="sample_unique_id", how="inner")


In [8]:
merged

Subject ID Patient Consented sample_unique_id   CGA  DOL   Current Weight  \
0      NB00012                 N      NB00012_M_1  28.4    27            1430   
1      NB00003                 Y      NB00003_M_1  29.0    22             995   
2      NB00003                 Y      NB00003_M_2  29.0    22            1005   
3      NB00003                 Y      NB00003_M_3  29.3    25            1065   
4      NB00354                 N      NB00354_M_1  29.4    12            1005   
..         ...               ...              ...   ...   ...             ...   
123    NB00007                 Y      NB00007_M_2  38.4    12            2190   
124    NB00007                 Y      NB00007_M_3  39.2    17            2340   
125    NB00007                 Y      NB00007_M_4  39.3    18            2380   
126    NB00379                 Y      NB00379_M_1  39.6    98            3073   
127    NB00007                 Y      NB00007_M_5  40.0    23            2430   

     Current Height  Current HC Scavenged/Fresh?   MBM/DMB?  ...      LNnT  \
0              32.0        26.0       Scavenged   MBM + DBM  ...  0.089623   
1              35.7        23.7       Scavenged         DBM  ...  0.054806   
2              36.0        24.4       Scavenged         DBM  ...  0.113728   
3              36.0        24.3       Scavenged         DBM  ...  0.078861   
4              34.0        34.5       Scavenged         MBM  ...  0.049258   
..              ...         ...              ...        ...  ...       ...   
123            44.0        32.5       Scavenged         MBM  ...  0.197148   
124            44.0        33.0       Scavenged         MBM  ...  0.093871   
125            44.0        33.0       Scavenged         MBM  ...  0.062626   
126            45.0        33.2       Scavenged         MBM  ...  0.074441   
127            45.0        34.5       Scavenged        MBM   ...  0.101503   

        LNFPI    LNFPII   LNFPIII      LSTc     DFLNT     DSLNT     DFLNH  \
0    0.470256  1.205369  0.014793  0.239572  0.281999  0.173245  0.215138   
1    0.337448  0.352980  0.011570  0.075870  0.375191  0.066067  0.044498   
2    0.508929  0.384073  0.009342  0.082319  0.296118  0.063142  0.073261   
3    0.579639  0.381617  0.012011  0.080428  0.181844  0.061159  0.052973   
4    0.108794  1.039603  0.024323  0.258750  0.138556  0.164200  0.203099   
..        ...       ...       ...       ...       ...       ...       ...   
123  1.031503  0.691480  0.038184  0.353092  0.681753  0.194094  0.063247   
124  0.826393  0.565923  0.023937  0.160066  0.732057  0.248158  0.140775   
125  0.250724  1.749928  0.062896  0.362996  0.159926  0.227344  0.067247   
126  1.492769  0.211561  0.009555  0.160130  0.005741  0.066606  0.061999   
127  0.756176  0.861343  0.024187  0.165676  0.767132  0.270879  0.059874   

       FDSLNH     DSLNH  
0    0.348744  0.384665  
1    0.038670  0.034095  
2    0.057761  0.057419  
3    0.090722  0.052794  
4    0.342770  0.364438  
..        ...       ...  
123  0.092453  0.154003  
124  0.036514  0.117773  
125  0.204275  0.226578  
126  0.013559  0.146605  
127  0.098085  0.144319  

[128 rows x 32 columns]

In [9]:
merged.columns

Index(['Subject ID', 'Patient Consented', 'sample_unique_id', 'CGA', 'DOL ',
       'Current Weight', 'Current Height', 'Current HC', 'Scavenged/Fresh?',
       'MBM/DMB?', 'HMF Y/N?', 'TPN Y/N?', 'Iron Y/N? ', 'Linked? ',
       '# Aliquots ', 'Additional Comments', 'inj vol', '2'FL', 'DFLAC',
       '3'SL', '6'SL', 'LNT', 'LNnT', 'LNFPI', 'LNFPII', 'LNFPIII', 'LSTc',
       'DFLNT', 'DSLNT', 'DFLNH', 'FDSLNH', 'DSLNH'],
      dtype='object')

* seeing 122 rows in the inner join - aka 122 samples that exist in both datasets
* seeing 135 rows in the left join - aka 135 samples that exist in the df dataset, fills unmatched HMO data with NaN

In [10]:
merged_2 = pd.merge(df, AC, on="sample_unique_id", how="left")
merged_2

Subject ID Patient Consented sample_unique_id   CGA  DOL   Current Weight  \
0      NB00012                 N      NB00012_M_1  28.4    27            1430   
1      NB00003                 Y      NB00003_M_1  29.0    22             995   
2      NB00003                 Y      NB00003_M_2  29.0    22            1005   
3      NB00003                 Y      NB00003_M_3  29.3    25            1065   
4      NB00354                 N      NB00354_M_1  29.4    12            1005   
..         ...               ...              ...   ...   ...             ...   
130    NB00007                 Y      NB00007_M_3  39.2    17            2340   
131    NB00007                 Y      NB00007_M_4  39.3    18            2380   
132    NB00379                 Y      NB00379_M_1  39.6    98            3073   
133    NB00007                 Y      NB00007_M_5  40.0    23            2430   
134    NB00007                 Y      NB00007_M_7  40.2    23            2400   

     Current Height  Current HC Scavenged/Fresh?   MBM/DMB?  ...      LNnT  \
0              32.0        26.0       Scavenged   MBM + DBM  ...  0.089623   
1              35.7        23.7       Scavenged         DBM  ...  0.054806   
2              36.0        24.4       Scavenged         DBM  ...  0.113728   
3              36.0        24.3       Scavenged         DBM  ...  0.078861   
4              34.0        34.5       Scavenged         MBM  ...  0.049258   
..              ...         ...              ...        ...  ...       ...   
130            44.0        33.0       Scavenged         MBM  ...  0.093871   
131            44.0        33.0       Scavenged         MBM  ...  0.062626   
132            45.0        33.2       Scavenged         MBM  ...  0.074441   
133            45.0        34.5       Scavenged        MBM   ...  0.101503   
134            45.0        34.5       Scavenged        MBM   ...       NaN   

        LNFPI    LNFPII   LNFPIII      LSTc     DFLNT     DSLNT     DFLNH  \
0    0.470256  1.205369  0.014793  0.239572  0.281999  0.173245  0.215138   
1    0.337448  0.352980  0.011570  0.075870  0.375191  0.066067  0.044498   
2    0.508929  0.384073  0.009342  0.082319  0.296118  0.063142  0.073261   
3    0.579639  0.381617  0.012011  0.080428  0.181844  0.061159  0.052973   
4    0.108794  1.039603  0.024323  0.258750  0.138556  0.164200  0.203099   
..        ...       ...       ...       ...       ...       ...       ...   
130  0.826393  0.565923  0.023937  0.160066  0.732057  0.248158  0.140775   
131  0.250724  1.749928  0.062896  0.362996  0.159926  0.227344  0.067247   
132  1.492769  0.211561  0.009555  0.160130  0.005741  0.066606  0.061999   
133  0.756176  0.861343  0.024187  0.165676  0.767132  0.270879  0.059874   
134       NaN       NaN       NaN       NaN       NaN       NaN       NaN   

       FDSLNH     DSLNH  
0    0.348744  0.384665  
1    0.038670  0.034095  
2    0.057761  0.057419  
3    0.090722  0.052794  
4    0.342770  0.364438  
..        ...       ...  
130  0.036514  0.117773  
131  0.204275  0.226578  
132  0.013559  0.146605  
133  0.098085  0.144319  
134       NaN       NaN  

[135 rows x 32 columns]

In [11]:
#list of IDs in the metadata but not the AC file

set(df['sample_unique_id']) - set(AC['sample_unique_id'])


{'NB00001_M_8',
 'NB00002_M_7',
 'NB00003_M_24',
 'NB00003_M_25',
 'NB00003_M_5',
 'NB00007_M_7',
 'NB00423_M_1'}

In [12]:
#list of IDs in the AC file but not in the metadata

set(AC['sample_unique_id']) - set(df['sample_unique_id'])


{'NB00001_M_2',
 'NB00002_M_9 ?',
 'NB00007_M_6',
 'NB00008_M_6',
 'NB00101_M_1',
 'NB00361_M_6',
 'NB00361_M_7'}

In [13]:
merged["Scavenged/Fresh?"].value_counts()

Scavenged/Fresh?
Scavenged     109
Scavenged      18
scavenged       1
Name: count, dtype: int64

In [14]:
def normalize_scavenged(value):
    if isinstance(value, str) and value.strip().lower().startswith('scavenged'):
        return 'Scavenged'
    return value

merged["Scavenged/Fresh?"] = merged["Scavenged/Fresh?"].apply(normalize_scavenged)

In [15]:
merged["Scavenged/Fresh?"].value_counts()

Scavenged/Fresh?
Scavenged    128
Name: count, dtype: int64

In [16]:
def code_mbm(value):
    if isinstance(value, str) and value.strip().lower().replace(' ', '') == 'mbm':
        return 'MBM'
    return value

merged["MBM/DMB?"] = merged["MBM/DMB?"].apply(code_mbm)

In [17]:
merged["MBM/DMB?"].value_counts()

MBM/DMB?
MBM                       92
DBM                       21
Switched to Fortifier      9
MBM + DBM                  2
FBM/MBM                    1
Name: count, dtype: int64

In [18]:
merged = merged.rename(columns={"Iron Y/N?\xa0": "Iron Y/N"})

In [19]:
merged["Iron Y/N"].value_counts()

Iron Y/N
Y    48
N    25
Name: count, dtype: int64

In [20]:
import re

def extract_numeric_aliquots(value):
    match = re.match(r'^\d+', str(value))
    return int(match.group()) if match else None

merged['Aliquots_num'] = merged['# Aliquots '].apply(extract_numeric_aliquots)

In [21]:
merged["Aliquots_num"].value_counts()

Aliquots_num
1     54
7     12
5     10
2      9
3      9
9      8
4      7
8      6
6      6
10     3
13     2
14     1
11     1
Name: count, dtype: int64

In [22]:
merged = merged.rename(columns={"2'FL": "2FL", "3'SL": "3SL", "6'SL": "6SL"})

In [23]:
merged["2FL"].describe()

count    126.000000
mean       2.668561
std        2.210266
min        0.034985
25%        0.300394
50%        2.904018
75%        4.311703
max        8.572130
Name: 2FL, dtype: float64

In [24]:
import plotly.express as px
fig = px.histogram(merged, x="2FL", nbins=40, title='2fl Concentration Distribution')
fig

In [25]:
# Classify secretor status using 5M AUC cutoff
merged["Secretor Status"] = merged["2FL"].apply(
    lambda x: "Secretor" if x >= .5 else "Non-Secretor"
)

In [26]:
merged["Secretor Status"].value_counts()

Secretor Status
Secretor        81
Non-Secretor    47
Name: count, dtype: int64

In [27]:
merged.columns

Index(['Subject ID', 'Patient Consented', 'sample_unique_id', 'CGA', 'DOL ',
       'Current Weight', 'Current Height', 'Current HC', 'Scavenged/Fresh?',
       'MBM/DMB?', 'HMF Y/N?', 'TPN Y/N?', 'Iron Y/N', 'Linked? ',
       '# Aliquots ', 'Additional Comments', 'inj vol', '2FL', 'DFLAC', '3SL',
       '6SL', 'LNT', 'LNnT', 'LNFPI', 'LNFPII', 'LNFPIII', 'LSTc', 'DFLNT',
       'DSLNT', 'DFLNH', 'FDSLNH', 'DSLNH', 'Aliquots_num', 'Secretor Status'],
      dtype='object')

In [28]:
merged["Additional Comments"].value_counts()

Additional Comments
Residual from Milk Prep Room                                                       68
Scavenged Feeding Tube                                                             48
Scavenged Feeding Tube + NaCl Oral Solution                                         2
Residual from Milk Prep Room & B2 Full                                              1
Scavenged Feeding Tube - Nurse needed to feed extra 1 mL                            1
Scavenged Feeding Tube - also started on sodium chloride oral solution 6/15/24      1
Scanvenged Feeding Tube - Collected Outside of 4 hr Window                          1
Scavenged Feeding Tube + NaCl Oral Solution - Pt intubated 6/25 & is NPO            1
Scavenged Feeding Tube - Outside 4-hr .indo. by 2 hours                             1
Scavenged Feeding Tube - Intubated 6/25 - NPO, restarted NG feeds 7/2               1
Scavenged Feeding Tube - Parenteral Nutrition Stopped                               1
Scavenged Feeding Tube - Last NG F

In [29]:
import re

def extract_scavenged_notes(comment):
    if isinstance(comment, str) and "scavenged feeding tube" in comment.lower():
        # Look for + or - and any text after it
        match = re.search(r"(scavenged feeding tube.*?)([+-].*)", comment, re.IGNORECASE)
        if match:
            return match.group(2).strip()
    return None

merged["scavenged notes"] = merged["Additional Comments"].apply(extract_scavenged_notes)

In [30]:
import re

def clean_scavenged_feeding_tube(comment):
    if isinstance(comment, str):
        # Find 'scavenged feeding tube' and remove everything after it (including +, -, or any text)
        match = re.search(r"(scavenged feeding tube)", comment, re.IGNORECASE)
        if match:
            return comment[:match.end()].strip()
    return comment

merged["Additional Comments"] = merged["Additional Comments"].apply(clean_scavenged_feeding_tube)

In [31]:
merged[["Additional Comments", "scavenged notes"]]

Additional Comments scavenged notes
0               Residual from Milk Prep Room & B2 Full            None
1                               Scavenged Feeding Tube            None
2                               Scavenged Feeding Tube            None
3                               Scavenged Feeding Tube            None
4                         Residual from Milk Prep Room            None
..                                                 ...             ...
123                       Residual from Milk Prep Room            None
124                       Residual from Milk Prep Room            None
125                       Residual from Milk Prep Room            None
126  Residual from Milk Prep Room - Baby not on NG ...            None
127                       Residual from Milk Prep Room            None

[128 rows x 2 columns]

In [32]:
# Remove leading '+', '-', or spaces from 'scavenged notes'
merged["scavenged notes"] = merged["scavenged notes"].str.lstrip('+- ').replace('', None)

In [33]:
merged["scavenged notes"].value_counts()

scavenged notes
NaCl Oral Solution                                       2
Nurse needed to feed extra 1 mL                          1
also started on sodium chloride oral solution 6/15/24    1
NaCl Oral Solution - Pt intubated 6/25 & is NPO          1
Outside 4-hr .indo. by 2 hours                           1
Intubated 6/25 - NPO, restarted NG feeds 7/2             1
Parenteral Nutrition Stopped                             1
Last NG Feed 7/2                                         1
Name: count, dtype: int64

In [34]:
merged["Additional Comments"].value_counts()

Additional Comments
Residual from Milk Prep Room                                            68
Scavenged Feeding Tube                                                  57
Residual from Milk Prep Room & B2 Full                                   1
Scanvenged Feeding Tube - Collected Outside of 4 hr Window               1
Residual from Milk Prep Room - Baby not on NG Feeds no linked sample     1
Name: count, dtype: int64

In [ ]:
def classify_sample_method(comment):
    if isinstance(comment, str):
        if "scavenged" in comment.lower():
            return "Scavenged"
        elif "milk prep room" in comment.lower():
            return "Residual"
    return "Other"

merged["Sample Source"] = merged["Additional Comments"].apply(classify_sample_method)


In [37]:
merged

Subject ID Patient Consented sample_unique_id   CGA  DOL   Current Weight  \
0      NB00012                 N      NB00012_M_1  28.4    27            1430   
1      NB00003                 Y      NB00003_M_1  29.0    22             995   
2      NB00003                 Y      NB00003_M_2  29.0    22            1005   
3      NB00003                 Y      NB00003_M_3  29.3    25            1065   
4      NB00354                 N      NB00354_M_1  29.4    12            1005   
..         ...               ...              ...   ...   ...             ...   
123    NB00007                 Y      NB00007_M_2  38.4    12            2190   
124    NB00007                 Y      NB00007_M_3  39.2    17            2340   
125    NB00007                 Y      NB00007_M_4  39.3    18            2380   
126    NB00379                 Y      NB00379_M_1  39.6    98            3073   
127    NB00007                 Y      NB00007_M_5  40.0    23            2430   

     Current Height  Current HC Scavenged/Fresh?   MBM/DMB?  ...      LSTc  \
0              32.0        26.0        Scavenged  MBM + DBM  ...  0.239572   
1              35.7        23.7        Scavenged        DBM  ...  0.075870   
2              36.0        24.4        Scavenged        DBM  ...  0.082319   
3              36.0        24.3        Scavenged        DBM  ...  0.080428   
4              34.0        34.5        Scavenged        MBM  ...  0.258750   
..              ...         ...              ...        ...  ...       ...   
123            44.0        32.5        Scavenged        MBM  ...  0.353092   
124            44.0        33.0        Scavenged        MBM  ...  0.160066   
125            44.0        33.0        Scavenged        MBM  ...  0.362996   
126            45.0        33.2        Scavenged        MBM  ...  0.160130   
127            45.0        34.5        Scavenged        MBM  ...  0.165676   

        DFLNT     DSLNT     DFLNH    FDSLNH     DSLNH  Aliquots_num  \
0    0.281999  0.173245  0.215138  0.348744  0.384665             5   
1    0.375191  0.066067  0.044498  0.038670  0.034095             2   
2    0.296118  0.063142  0.073261  0.057761  0.057419             1   
3    0.181844  0.061159  0.052973  0.090722  0.052794             1   
4    0.138556  0.164200  0.203099  0.342770  0.364438             7   
..        ...       ...       ...       ...       ...           ...   
123  0.681753  0.194094  0.063247  0.092453  0.154003             7   
124  0.732057  0.248158  0.140775  0.036514  0.117773             6   
125  0.159926  0.227344  0.067247  0.204275  0.226578             7   
126  0.005741  0.066606  0.061999  0.013559  0.146605             7   
127  0.767132  0.270879  0.059874  0.098085  0.144319             2   

     Secretor Status  scavenged notes  Sample Source  
0           Secretor             None       Residual  
1           Secretor             None      Scavenged  
2           Secretor             None      Scavenged  
3           Secretor             None      Scavenged  
4       Non-Secretor             None       Residual  
..               ...              ...            ...  
123         Secretor             None       Residual  
124         Secretor             None       Residual  
125     Non-Secretor             None       Residual  
126         Secretor             None       Residual  
127         Secretor             None       Residual  

[128 rows x 36 columns]

In [36]:
merged.to_excel('Unlinked_Merged.xlsx', index=False)

### is HMO concentration accosicated with metadata features?

In [ ]:
pip install scipy


In [ ]:
merged

In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt
from scipy.stats import mannwhitneyu

sns.boxplot(data=merged, x="TPN Y/N?", y="2FL")
plt.title("2’FL Concentration by TPN Use")
plt.show()

# Mann-Whitney U test (non-parametric)
group1 = merged[merged["TPN Y/N?"] == "Y"]["2FL"].dropna()
group2 = merged[merged["TPN Y/N?"] == "N"]["2FL"].dropna()

stat, p = mannwhitneyu(group1, group2)
print(f"Mann-Whitney U test: U={stat:.2f}, p={p:.4f}")


### Only with Subject IDs with more than 3 timepoints

In [ ]:
subject_counts = merged["Subject ID"].value_counts()
longitudinal_subjects = subject_counts[subject_counts > 3].index

filtered_df = merged[merged["Subject ID"].isin(longitudinal_subjects)].copy()

In [ ]:
filtered_df.columns = filtered_df.columns.str.strip()


In [ ]:
filtered_df["MBM/DMB?"]

In [ ]:
filtered_df.columns

In [ ]:
df = merged.rename(columns=lambda x: x.strip())  # Remove trailing spaces
df["DOL"] = pd.to_numeric(df["DOL"], errors="coerce")

In [ ]:
import plotly.express as px

fig = px.line(
    df[df["Subject ID"] == 'NB00003'],  # set selected_subject as needed
    x="DOL",
    y="2FL",
    color="TPN Y/N?",  # or "TPN Y/N?"
    markers=True,
    title=f"2’FL over Time for {'NB00003'} by HMF Status"
)
fig.update_layout(
    plot_bgcolor="#1E1E1E",
    paper_bgcolor="#1E1E1E",
    font_color="white"
)
fig.show()


In [ ]:
from scipy.stats import mannwhitneyu

group1 = df[df["TPN Y/N?"] == "Y"]["2FL"].dropna()
group2 = df[df["TPN Y/N?"] == "N"]["2FL"].dropna()

stat, p = mannwhitneyu(group1, group2)
print(f"TPN Effect on 2’FL: U = {stat:.2f}, p = {p:.4f}")


In [ ]:
df.columns

In [ ]:
df["hover_label"] = (
    "TPN: " + df["TPN Y/N?"].astype(str) + "<br>" +
    "HMF: " + df["HMF Y/N?"].astype(str) + "<br>" +
    "Iron: " + df["Iron Y/N"].astype(str) + "<br>" +
    "Scavenged: " + df["Scavenged/Fresh?"].astype(str) + "<br>" +
    "Notes: " + df["Additional Comments"].astype(str)
)

In [ ]:
import plotly.express as px

In [ ]:
df["Iron Size"] = df["Iron Y/N"].map({"Y": 20, "N": 10})


In [ ]:
import plotly.express as px

fig = px.scatter(
    df[df["Subject ID"] == "NB00003"],
    x="DOL",
    y="2FL",
    color="TPN Y/N?",
    symbol="Scavenged/Fresh?",
    size="Iron Size",
    hover_name="DOL",
    hover_data={
        "TPN Y/N?": True,
        "HMF Y/N?": True,
        "Iron Y/N": True,
        "Scavenged/Fresh?": True,
        "Additional Comments": True
    },
    text="HMF Y/N?",
    title="2’FL over Time for NB00003"
)


fig.update_layout(
    plot_bgcolor="#1E1E1E",
    paper_bgcolor="#1E1E1E",
    font_color="white"
)

fig.show()


## Type of Milk Collected

In [ ]:
merged["Additional Comments"].value_counts()

In [ ]:
filtered_merged = merged[
    merged["Additional Comments"].isin(["Residual from Milk Prep Room", "Scavenged Feeding Tube"])
].copy()

In [ ]:
filtered_merged

In [ ]:
def classify_sample_method(comment):
    if isinstance(comment, str):
        if "scavenged feeding tube" in comment.lower():
            return "Scavenged"
        elif "milk prep room" in comment.lower():
            return "Residual"
    return "Other"

filtered_merged["Sample Source"] = filtered_merged["Additional Comments"].apply(classify_sample_method)


In [ ]:
filtered_merged["Sample Source"].value_counts()

In [ ]:
filtered_merged["CGA"].describe()

In [ ]:
filtered_merged["CGA Bin"] = pd.cut(
    filtered_merged["CGA"], 
    bins=[28, 32, 34, 36, 41], 
    labels=["Very Preterm (<32)", "Moderate Preterm (32-34)", "Late Preterm (34–36)", "Term (35–40)"]
)


In [ ]:
filtered_merged[""]